modules/bibdocfile/lib/bibdocfile.py

```diff
@@ -3874,9 +3874,12 @@ def download_external_url(url, docformat=None):
     from_file, to_file, tmppath = None, None, ''
 
     try:
-        while r.get('hostname_cache.%s' % hostname):
-            # We have accessed this host within the last 2 seconds
-            time.sleep(2)
+        if hostname == CFG_LABS_HOSTNAME:
+            time.sleep(0.1)
```

Comment:
is the `sleep` needed at all?

Reply:
I don't know, I asked twice on ops channel about this. I have no odea how loaded/busy labs api nodes can be. I do occasionally see timeouts from callbacks